include/mbedtls/pk.h

```diff
@@ -535,6 +535,45 @@ int mbedtls_pk_sign( mbedtls_pk_context *ctx, mbedtls_md_type_t md_alg,
              unsigned char *sig, size_t sig_size, size_t *sig_len,
              int (*f_rng)(void *, unsigned char *, size_t), void *p_rng );
 
+/**
+ * \brief           Make signature with options, including padding if relevant.
+ *
+ * \param type      Signature type (inc. possible padding type) to verify
+ * \param options   Pointer to type-specific options, or NULL
+ * \param ctx       The PK context to use. It must have been set up
+ *                  with a private key.
+ * \param md_alg    Hash algorithm used (see notes)
+ * \param hash      Hash of the message to sign
+ * \param hash_len  Hash length
+ * \param sig       Place to write the signature.
+ *                  It must have enough room for the signature.
+ *                  #MBEDTLS_PK_SIGNATURE_MAX_SIZE is always enough.
+ *                  You may use a smaller buffer if it is large enough
+ *                  given the key type.
+ * \param sig_size  The size of the \p sig buffer in bytes.
+ * \param sig_len   On successful return,
+ *                  the number of bytes written to \p sig.
+ * \param f_rng     RNG function, must not be \c NULL.
+ * \param p_rng     RNG parameter
+ *
+ * \return          0 on success, or a specific error code.
+ *
+ * \note            For RSA keys, the default padding type is PKCS#1 v1.5.
+ *                  There is no interface in the PK module to make RSASSA-PSS
```

Comment:
Actually, there, is: this very function :) Suggested replacement for this note: `For RSA keys, the padding type depends on the value of the \p type parameter: MBEDTLS_PK_RSA for PKCS#1 v1.5, and MBEDTLS_PK_RSASSA_PSS for PKCS#1 v2.1.`

If going with my suggestion of directly going for a PSA-based solution (see comment on the implementation), then we'll add `MBEDTLS_PK_RSASSA_PSS is only supported if MBEDTLS_PSA_CRYPTO_C is defined; the signature uses MGF1 with the hash given by \p md_alg and a salt length equal to the hash length`.

We also need to document how `options` is used. With the current implementation, they can be used, but this would be a problem when doing the PSA version, as PSA doesn't support options. So, I think we can save ourselves some trouble and directly document that `\p options is currently ignored for all supported algorithms and must be NULL`. Doing otherwise is creating backwards compatibility problems for out future selves.

(Note: what PSA does is what TLS 1.3 mandates, so we definitely don't need options for TLS 1.3.)

Reply:
I removed `options ` and add `#if defined(MBEDTLS_USE_PSA_CRYPTO) || defined(MBEDTLS_SSL_PROTO_TLS1_3)` to guard the function.